pathogen-workflows.sql

```diff
@@ -191,12 +220,92 @@ run as materialized (
     where
             age(run.created_at) <= '90 days'
         and run.created_at >= '2024-02-13T21:50:28Z'::timestamptz -- When I merged <https://github.com/nextstrain/.github/pull/54>. —trs
+),
+
+/* Filter to the last 30 runs.
+ */
+recent as materialized (
+    select
+        *
+    from
+        run_last_attempt
+    where
+        relative_workflow_run_number <= 30
+),
+
+/* Backfill previous run attempts.
+ */
+attempt as materialized (
+    /* If the last run is attempt 1, then we can use the row we already have
+     * as-is without fetching additional rows.  This is true for most runs!
+     * That's good, because GitHub's REST API doesn't provide an efficient way
+     * of fetching previous run attempts.
+     *   -trs, 17 Dec 2024
+     */
+    select
+        repository_full_name,
+        workflow_id,
+        workflow_path,
+        workflow_name,
+        workflow_html_url,
+        workflow_run_number,
+        id,
+        attempt,
+        html_url,
+        created_at,
+        updated_at,
+        conclusion,
+        status,
+        duration,
+        event,
+        commit_id,
+        commit_msg
+    from
+        recent
+    where
+        attempt = 1
+
+    union all
+
+    select
+        recent.repository_full_name,
+        recent.workflow_id,
+        recent.workflow_path,
+        recent.workflow_name,
+        recent.workflow_html_url,
+        recent.workflow_run_number,
+        recent.id,
+
+        -- Keep in sync with the matching select list in "run_last_attempt" above.
+        run.run_attempt                 as attempt,
+        format('%s/attempts/%s', run.html_url, run.run_attempt)
+                                        as html_url,
+        run.created_at,
+        run.updated_at,
+        run.conclusion,
+        run.status,
+        case when run.conclusion is not null
+            then run.updated_at - run.created_at
+            else current_timestamp(0) - run.created_at
+        end                             as duration,
+        run.event,
```

Comment:
Slightly weird that the event is still the original event (e.g. "schedule") for later attempts. Thoughts on hardcoding `re-run` or `format('%s/re-run', run.event)` here? 

Hmm, maybe the original event would make more sense if the `attempt` number is displayed next to the event in the UI.

Reply:
Oh! Yes, totally should surface that somehow. That's a complete oversight on my part. Too much focus on just _getting_ the data. :facepalm:

I know you know this, but to clarify for anyone else reading: GitHub uses the exact same event context on re-runs, which is really important and makes sense, but is also weird/confusing unless additional information is surfaced like the attempt number or who triggered the re-run.

Reply:
9a40ae7

![image](https://github.com/user-attachments/assets/395cfc30-adc0-4910-a348-e5cdcc6786a5)